df_to_vec.py

```diff
@@ -0,0 +1,199 @@
+import random
+import time
+
+import pandas as pd
+from gensim.models import Word2Vec
+import numpy as np
+import os
+
+WORD_VEC_LENGTH = 100
+NUMBER_OF_TYPES = 1000
+
+w2v_models = {
+    'code': Word2Vec.load('resources/word2vec_model_code.bin'),
+    'language': Word2Vec.load('resources/word2vec_model_language.bin')
+}
+
+
+# NL2Type code
+def vectorize_string(text, feature_length, w2v_model):
+    text_vec = np.zeros((feature_length, WORD_VEC_LENGTH))
+    if text == 'unknown':
+        return text_vec
+    count = 0
+    for word in text.split():
+        if count >= feature_length:
+            return text_vec
+        try:
+            text_vec[count] = w2v_model.wv[word]
+        except KeyError:
+            pass
+        count += 1
+
+    return text_vec
+# /NL2Type code
+
+class Datapoint:
+    def __repr__(self) -> str:
+        values = list(map(lambda kv: kv[0] + ': ' + repr(kv[1]), self.__dict__.items()))
+        values = "\n\t" + ",\n\t".join(values) + "\n"
+        return type(self).__name__ + "(%s)" % values
+
+    def vector_length(self):
+        return sum(self.features.values()) + len(self.features.values()) - 1
+
+    def to_vec(self):
+        datapoint = np.zeros((self.vector_length(), WORD_VEC_LENGTH))
+
+        separator = np.ones(WORD_VEC_LENGTH)
+
+        position = 0
+        for feature, feature_length in self.features.items():
+
+            if self.feature_types[feature] == 'datapoint_type':
```

Comment:
Is it possible that a DataPoint does not have a `datapoint_type`? .If you can assume this, then you could remove it from the dict and call this whats inside this if block above the for loop.

Reply:
No, all datapoints _should_ have a type. However, I'd like to keep it in the feature lengths dict. Therefore it makes more sense to do it within the for loop than creating an exception for this case within the for loop and doing the actual work outside the for loop.